### DISCUSSION TITLE
How many checks?

### DISCUSSION THREAD

**Discussion by iadawn** [w3c/wai-easy-checks#145] [Ideas]
How many checks is too many?

Some other ideas around this...

One options is to group checks on the resource homepage. This would allow for more to be included without it seeming too overwhelming.

Grouping could be by theme or by difficulty.

**Comment by shawna-slh** [comment]
+100 for grouping them.

Maybe:
1. ~7 main checks
2. Additional really easy checks (expandable or separate page)
3. More difficult checks (expandable or separate page)

And maybe grouping by theme within there?

And probably grouping and/or clear indication of:
* Specific checks
  * for forms
  * for audio and video
  * other?

... not sure how that relates to the grouping above, though :-)

Some specifics in "shawn" column of [Easy Checks - Prioritization (google sheet)](https://docs.google.com/spreadsheets/d/1FsKGXi1BrBHBId_ZwKLtloXBZL0GcouQ7c26r-XJLRw/edit#gid=0)

**Comment by iadawn** [comment]
Just to flag a comment I made in the [EO meeting on 9th June](https://www.w3.org/2023/06/09-eo-minutes.html#item02); having a small number isn't necessarily the important thing. Ensuring that the overview page doesn't become overwhelming is the thing. This is where grouping will be useful.

**Comment by shawna-slh** [comment]
Should we include all the the easy checks that were in the [previous version](https://www.w3.org/WAI/test-evaluate/preliminary/)? (OK to leave off the harder ones?)

**Comment by shawna-slh** [comment]
OK here's a pass at one idea to "get the brain juices flowing" **for editors discussion then EOWG discussion**.

**Background: [Easy Checks Prioritization](https://docs.google.com/spreadsheets/d/1FsKGXi1BrBHBId_ZwKLtloXBZL0GcouQ7c26r-XJLRw/edit?pli=1#gid=0)**

<hr>

### Easy checks:
* [Language](https://deploy-preview-131--wai-easychecks.netlify.app/test-evaluate/easy-checks/language/)
* [Page title](https://deploy-preview-131--wai-easychecks.netlify.app/test-evaluate/easy-checks/title/)
* [Orientation for apps](https://github.com/w3c/wai-easy-checks/blob/2023-update/content/orientation.md)

### Core checks:
* Image text alternatives ("alt text") (pictures, illustrations, charts, etc.)
* [Headings](https://deploy-preview-131--wai-easychecks.netlify.app/test-evaluate/easy-checks/heading-structure/)
* Text contrast ratio ("color contrast")
* [Moving, Flashing, or Blinking Content](https://deploy-preview-131--wai-easychecks.netlify.app/test-evaluate/easy-checks/moving/)
* [Text spacing](https://deploy-preview-131--wai-easychecks.netlify.app/test-evaluate/easy-checks/text-spacing/) (or, maybe include with zoom since you’re looking for the same thing)

### Video and audio checks:
* [Captions](https://deploy-preview-131--wai-easychecks.netlify.app/test-evaluate/easy-checks/captions/)
* [Transcripts](https://deploy-preview-131--wai-easychecks.netlify.app/test-evaluate/easy-checks/transcripts/)
* [Description](https://deploy-preview-131--wai-easychecks.netlify.app/test-evaluate/easy-checks/description/)
* [Moving, Flashing, or Blinking Content](https://deploy-preview-131--wai-easychecks.netlify.app/test-evaluate/easy-checks/moving/)

### Harder checks:
* Resize text, zoom, reflow _{@@ ? together or separate ? too complex together? or too repetitive separate?}_
* Keyboard access and visual focus _(maybe + focus not obscured) {@@ ? together or separate ? too complex together? or too repetitive separate?}_
* Form field labels

<hr>

_**Maybe add later:**_
* ? Dragging movements - wasn’t in the priorities spreadsheet
* Focus not obscured OR maybe include as a check with keyboard access?
* Links open new windows – it’s AAA
* Skip links – it’s not required in some cases. Would it be too cumbersome to explain that?
* Lists – because it was low difficulty to explain and low difficulty to perform we might want to include it even though it's also low impact in many situations

_**Not including in this iteration:**_
* ? Reading level - AAA - while tools help, they have limitations
* Link text – when we worked on the first version, we decided this was prone to false negatives. Can revisit, but would be good to dig up issues from the to avoid re-inventing or spinning
* Required fields
* Redundant color coding – not common
* Focus order – no one argued for inclusion
* Landmarks – no one argued for inclusion
* Error handling – no one argued for inclusion
* Tables – no one argued for inclusion
* Consistent help – no one argued for inclusion
* Non-text contrast – no one argued for inclusion
* Target size – no one argued for inclusion
* Redundant entry – no one argued for inclusion
* Accessible authentication – no one argued for inclusion
* Animation from interaction – no one argued for inclusion

@iadawn @AndrewArch

**Comment by AndrewArch** [comment]
Shawn, I'm happy with your grouping & prioritisation for now. I would personally like to see readability added back in - could even be an easy check if [Hemingway](https://hemingwayapp.com/) is accessible

**Comment by iadawn** [comment]
Couple of things that I would argue about:

* Keeping each check as atomic as possible - for example, keyboard access and visible focus checks two different (albeit related items). Grouping them in the easy check makes it more involved to explain and unpack.
* Easy checks vs core checks - not sure on the terminology
* I would not include forms as a 'harder' check but as a specific group - they are too ubiquitous a barrier not to touch on with some degree of focus
* I would prefer to avoid any mention of something being 'harder' - will tend to make people think that and either avoid or struggle with the thing

**Comment by shawna-slh** [comment]
To clarify -- [one idea above](https://github.com/w3c/wai-easy-checks/issues/140#issuecomment-1673976099) was just to have something concrete to discuss &mdash; it was not meant to be a proposal at all. I added text to hopefully make more clear.

**Comment by AndrewArch** [comment]
On reflection I'm wondering if we try and group the checks by what people will be checking, e.g.:
* Structure
* Navigation
* Multimedia
* Forms
* etc. 

We might end up with a miscellaneous group at the end, but I think this type of grouping would be better than Easy, Core, Harder mixed in with Multimedia and other information-type groupings

**Comment by shawna-slh** [comment]
***If*** all the checks can be made easy, then I fully support that idea.

My *big* concern is how often we've heard, "this is not easy" -- from users in hands-on WAI tour sessions to recent usability testing of the prototypes. (a few notes about that in https://github.com/w3c/wai-easy-checks/issues/84 )

My gut feeling is that we'll want to ease users in with some really easy checks first. And, we'll probably want to indicate some that might be harder, in respect for the audience.

And, I'm quite happy to be wrong -- that we craft all checks to be pretty easy to most audiences! :-)

(And even if we did re-title it, I think we still have the same [purpose](https://www.w3.org/WAI/EO/wiki/Easy_Checks_2022#Purpose), [audiences](https://www.w3.org/WAI/EO/wiki/Easy_Checks_2022#Target_Audience), [use cases](https://www.w3.org/WAI/EO/wiki/Easy_Checks_2022#Use_Cases), etc.)
(note not to let this issue tangent into the title discussion - if we want to revisit that, it should be a new issue or discussion.)

**Reply by iadawn** [reply]
I think we could do both: group by thing to be checked but also indicate how 'involved' or 'easy' a check is. We are playing with a design idea that would allow for this:

<img width="499" alt="image" src="https://github.com/w3c/wai-easy-checks/assets/2444840/21d67ec6-c393-4533-b213-c7f6a42b96d7">

So have a group of checks but in the check indicate some meta information about the check. For example, roughly how long should the check take, how technical is it, how easy is it, who is best placed to do the check.

This approach helps to address both issues.

**Comment by shawna-slh** [comment]
I also agree that we don't need to make final decisions on grouping for a while.

One idea for approach:
* Finish up quick drafts of the checks that we will probably include in the first iteration (but not spend much time on the ones we're not sure EOWG will want to include)
* Bring those to EOWG for [eagle review](https://www.w3.org/WAI/EO/wiki/EOWG_Participation_Info#Eagle_Review_.E2.80.94_Bird.27s-eye_view_.2F_30.2C000-feet) along with questions on potential grouping.

**Comment by shawna-slh** [comment]
>  For example, roughly how long should the check take, how technical is it, how easy is it, who is best placed to do the check.
> Content creators, 20 minutes, high impact

< wet blanket alert >

I love the concept in general … and I’m not sure how it will work when we get to the details. A few points:

1. re: “who is best placed to do the check” and “Content creators “ -- Please review the [purpose](https://www.w3.org/WAI/EO/wiki/Easy_Checks_2022#Purpose), [audiences](https://www.w3.org/WAI/EO/wiki/Easy_Checks_2022#Target_Audience), and especially the [use cases](https://www.w3.org/WAI/EO/wiki/Easy_Checks_2022#Use_Cases). I don’t think “who is best placed to do the check” is relevant for most of the use cases.

2. re: “roughly how long should the check take” – Challenge is that there is so much variation in time.
* When you are learning a check, it might take 30+ minutes. Then, once you know how to do it, it might take 3 minutes.
* Several checks depend on the content of a page. E.g., is there 1 image or 25 images on the page?
* Some people are slow. Some are fast. I thought of this comment from [Len]( https://www.w3.org/WAI/EO/wiki/EOWG_2023#Deliverables_Brainstorming) “Can we add some 'phrases to avoid'… For example, "simply fill out this quick form" is ableist and can set the wrong tone for the user's ability to complete the task.”

3. “high impact” While I think this was good for our internal analysis of which checks to include, it could be difficult or problematic to include impact in the resource itself – because the impact on an individual in a specific situation could be very different. E.g., something that is usually low impact for most PWDs could be high impact for an individual person on a specific page.

**Comment by shawna-slh** [comment]
A related thing: Might we want to indicate which checks require sight? For some checks, you'll need to visually see the page to do some or all of the check.

**Comment by mpaiva** [comment]
![IMG_7976](https://github.com/w3c/wai-easy-checks/assets/781670/c32fea85-a187-4bf2-a6a2-0e7eb197ff82)

**Reply by mpaiva** [reply]
![IMG_7977](https://github.com/w3c/wai-easy-checks/assets/781670/b4f76c11-2324-4994-80e7-a77ceb842882)

**Reply by shawna-slh** [reply]
description of images - cards labeled and sorted:
1. https://docs.google.com/spreadsheets/d/1FsKGXi1BrBHBId_ZwKLtloXBZL0GcouQ7c26r-XJLRw/edit?pli=1#gid=1797365397
2. https://docs.google.com/spreadsheets/d/1FsKGXi1BrBHBId_ZwKLtloXBZL0GcouQ7c26r-XJLRw/edit?pli=1#gid=1765696469
